src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -74,6 +76,7 @@ public function getData($id)
             'shipping' => $this->extractShippingData($productForEditing),
             'options' => $this->extractOptionsData($productForEditing),
             'suppliers' => $this->extractSuppliersData($productForEditing),
+            'customizations' => $this->extractCustomizationsData($productForEditing),
```

Comment:
So, now that I think of it, `public function getData($id)` retrieves a `GetProductForEditing` result and map it to an array, so that next it can be put inside a form?

Reply:
yes thats the idea of each formProvider, it retrieves the data for existing identifiable object and puts it in form inputs edit forms